application/models/TemplateConfiguration.php

```diff
@@ -1354,4 +1380,188 @@ public function setOptionKeysToInherit(){
+     */
+    public function getVirtualThemeFilePath($path)
+    {
+        /** @var LimeSurvey\Datavalueobjects\ThemeFileInfo|null */
+        $fileInfo = $this->getThemeFileInfo($path);
+
+        if (empty($fileInfo)) {
+            return null;
+        }
+
+        return $fileInfo->virtualPath;
+    }
+
+    /**
+     * Returns the real absolute path of $path if it's valid.
+     *
+     * @param string $path  the path to check. Can be a "virtual" path (eg. 'image::theme::logo.png'), or a normal path.
+     *
+     * @return string|null the real absolute path if it's valid, of null if it's not.
+     */
+    public function getRealThemeFilePath($path)
+    {
+        /** @var LimeSurvey\Datavalueobjects\ThemeFileInfo|null */
+        $fileInfo = $this->getThemeFileInfo($path);
+
+        if (empty($fileInfo)) {
+            return null;
+        }
+
+        return $fileInfo->realPath;
+    }
+
+    /**
+     * Returns the theme file information if the path is valid.
+     * To be valid, the $path must correspond to one of the theme file categories.
+     *
+     * @param string $path  the path to check. Can be a "virtual" path (eg. 'image::theme::logo.png'), or a normal path.
+     *
+     * @return LimeSurvey\Datavalueobjects\ThemeFileInfo|null the file info if it's valid, of null if it's not.
+     */
+    public function getThemeFileInfo($path)
+    {
+        if (is_null($path)) {
+            return null;
+        }
+
+        /** @var LimeSurvey\Datavalueobjects\ThemeFileCategory[] */
+        $categoryList = $this->getFileCategories();
+
+        // Check if the path matches the virtual path category format
+        $prefix = $this->getVirtualPathPrefix($path);
+        if (!empty($prefix)) {
+            $filteredCategories = array_filter($categoryList, function ($v) use ($prefix) { return $v->pathPrefix == $prefix; });
+            if (empty($filteredCategories)) {
+                return null;    // No category matched the path's prefix
+            }
+            $category = reset($filteredCategories);
+
+            // Validate that the file exists
+            $realPath = realpath($category->path . '/' . substr($path, strlen($prefix)));
+
+            // If the file exists and no traversing is done (the real path starts with the category's base path),
+            // return the real path
+            if ($realPath !== false && strpos($realPath, $category->path) === 0) {
+                $virtualPath = str_replace($category->path, $category->pathPrefix, $realPath);
+                return new ThemeFileInfo($realPath, $virtualPath, $category);
+            } else {
+                return null;
+            }
+        }
+
+        // Path doesn't match the virtual path category format, so we try the determine if it belongs to a category.
+        foreach ($categoryList as $category) {
+            // Check for paths relative to the theme, relative to the rootdir, or absolute
```

Comment:
Get the realpath for the file. Test whether it is relative to the category, rootdir or absolute